cmd/watch.go

```diff
@@ -11,17 +11,16 @@ import (
 	"gotest.tools/gotestsum/testjson"
 )
 
-func runWatcher(opts *options) error {
-	ctx, cancel := context.WithCancel(context.Background())
-	defer cancel()
-
-	w := &watchRuns{opts: *opts}
+func runWatcher(ctx context.Context, cancelFunc context.CancelFunc, opts *options) error {
+	w := &watchRuns{opts: *opts, ctx: ctx, cancel: cancelFunc}
 	return filewatcher.Watch(ctx, opts.packages, opts.watchClear, w.run)
 }
 
 type watchRuns struct {
 	opts     options
 	prevExec *testjson.Execution
+	ctx      context.Context
+	cancel   context.CancelFunc
 }
```

Comment:
Wait, there is something strange.

Context should never be embedded in a struct.

I feel like there is something strange

Reply:
My point was to pass ctx as first argument of this method

Reply:
Ok fixed :)